javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionConstructorDeclaration.java

```diff
@@ -47,8 +46,12 @@ public ReflectionConstructorDeclaration(Constructor<?> constructor, TypeSolver t
     }
 
     @Override
-    public ResolvedClassDeclaration declaringType() {
-        return new ReflectionClassDeclaration(constructor.getDeclaringClass(), typeSolver);
+    public ResolvedReferenceTypeDeclaration declaringType() {
```

Comment:
An API change is required here since the declaring type for a constructor can be a record as well